### ISSUE TITLE
Setting LD_LIBRARY_PATH kills dotnet command

### ISSUE THREAD

**Issue by isaacabraham** [dotnet/cli#9936]
Trying to run a dotnet application on a Ubuntu machine with MatLab installed. Note - I'm not suggesting that this in itself is a bug with dotnet itself, but I would appreciate if someone can shed any light on why setting this environment variable kills the SDK.

## Steps to reproduce
1. Create an Ubuntu VM (16.04 here).
1. Install the .NET SDK (2.1.401 here).
1. Run `dotnet --info` (works fine).
1. Apply the following command:

```sh
MCRROOT=/usr/local/MATLAB/MATLAB_Runtime/v90
LD_LIBRARY_PATH=.:${MCRROOT}/runtime/glnxa64 ;
LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:${MCRROOT}/bin/glnxa64 ;
LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:${MCRROOT}/sys/os/glnxa64;
LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:${MCRROOT}/sys/opengl/lib/glnxa64;
export LD_LIBRARY_PATH;
```

5. Run `dotnet --info` again. The following error is displayed:

```sh
dotnet: /usr/local/MATLAB/MATLAB_Runtime/v90/sys/os/glnxa64/libstdc++.so.6: version `GLIBCXX_3.4.18' not found (required by dotnet)
```

## Expected  behavior
dotnet continues to run.

## Actual behavior
dotnet does not run.

## Environment data
`dotnet --info` output:
```.NET Core SDK (reflecting any global.json):
 Version:   2.1.401
 Commit:    91b1c13032

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  16.04
 OS Platform: Linux
 RID:         ubuntu.16.04-x64
 Base Path:   /usr/share/dotnet/sdk/2.1.401/

Host (useful for support):
  Version: 2.1.3
  Commit:  124038c13e

.NET Core SDKs installed:
  2.1.401 [/usr/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.3 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.3 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.3 [/usr/share/dotnet/shared/Microsoft.NETCore.App]```

**Comment by livarcocc** [comment]
@jkotas would you know?

**Comment by jkotas** [comment]
`LD_LIBRARY_PATH` overrides paths to system libraries. If the overriden libraries are not compatible with the system libraries, you will see crashes. 

There is a number of articles that talk about dangers of using `LD_LIBRARY_PATH`, e.g. from http://wiredrevolution.com/system-administration/how-to-correctly-use-ld_library_path :

> In general it is not a good practice to have `LD_LIBRARY_PATH` permanently set in your environment. This could lead to unintended side effects as programs can link to unintended libraries producing strange results or unexpectedly crashing. There is also the possibility introducing potential security threats.

It is what's happening here.

cc @janvorli

**Comment by janvorli** [comment]
Right, in this case, the /usr/local/MATLAB/MATLAB_Runtime/v90/sys/os/glnxa64/ contains an obsolete version of libstc++. So all applications written in C++ that are launched with the LD_LIBRARY_PATH as shown above will get this old libstdc++. For some applications, it may be ok, but for applications that depend on functions from a newer libstdc++ like our dotnet tool, it inevitably leads to crashes.

**Comment by isaacabraham** [comment]
Thanks very much for getting back to me @jkotas @janvorli . This leaves the question that this *appears* to be more or less official guidance for installing MatLab. Does this mean that there's an incompatibility between MatLab and the dotnet CLI? Can one leave them out without issue?

**Comment by janvorli** [comment]
@isaacabraham can you please point me to the installation instructions for MATLAB that mention the LD_LIBRARY_PATH setting?

**Comment by isaacabraham** [comment]
Here's [one](https://uk.mathworks.com/help/matlab/matlab_external/set-run-time-library-path-on-linux-systems.html). Also, after you install matlab itself, the actual installation script prints out the instructions to apply to it. I'll see if I can get you a copy of them later today.

**Comment by dasMulli** [comment]
> Set the path every time you run MATLAB. Alternatively, place the commands in a MATLAB startup script.

at least they don't ask to make this your system default as it only applies to Matlab.

**Comment by janvorli** [comment]
I agree, the guidance to put it into the .profile / .cshrc is really not nice.
@isaacabraham I would recommend creating a .sh file that would contain the LD_LIBRARY_PATH setting and matlab launching instead. You would use this script to launch matlab and your environment would stay clean for other applications like dotnet.

**Comment by isaacabraham** [comment]
We're actually doing this now from within our .NET app by spawning a new process and setting the environment variable there. But yes, isolation seems the correct approach here. Thanks for helping out.
